src/main/java/org/wso2/carbon/inbound/vfs/FileInjectHandler.java

```diff
@@ -0,0 +1,243 @@
+import org.apache.axis2.builder.SOAPBuilder;
+import org.apache.axis2.context.MessageContext;
+import org.apache.axis2.format.DataSourceMessageBuilder;
+import org.apache.axis2.format.ManagedDataSource;
+import org.apache.axis2.format.ManagedDataSourceFactory;
+import org.apache.axis2.transport.TransportUtils;
+import org.apache.commons.io.input.AutoCloseInputStream;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.apache.synapse.SynapseConstants;
+import org.apache.synapse.SynapseException;
+import org.apache.synapse.commons.vfs.VFSConstants;
+import org.apache.synapse.commons.vfs.VFSOutTransportInfo;
+import org.apache.synapse.core.SynapseEnvironment;
+import org.apache.synapse.inbound.InboundEndpoint;
+import org.apache.synapse.mediators.base.SequenceMediator;
+import org.apache.synapse.transport.customlogsetter.CustomLogSetter;
+import org.wso2.org.apache.commons.vfs2.FileObject;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.Map;
+
+import javax.mail.internet.ContentType;
+import javax.mail.internet.ParseException;
+
+import static org.apache.axis2.context.MessageContext.TRANSPORT_HEADERS;
+
+public class FileInjectHandler {
+
+    private static final Log log = LogFactory.getLog(FileInjectHandler.class);
+
+    private String injectingSeq;
+    private String onErrorSeq;
+    private boolean sequential;
+    private VFSConfig vfsProperties;
+    private SynapseEnvironment synapseEnvironment;
+    private Map<String, Object> transportHeaders;
+    private String fileURI;
+
+    public FileInjectHandler(String injectingSeq, String onErrorSeq, boolean sequential,
+                             SynapseEnvironment synapseEnvironment, VFSConfig vfsProperties) {
+        this.injectingSeq = injectingSeq;
+        this.onErrorSeq = onErrorSeq;
+        this.sequential = sequential;
+        this.synapseEnvironment = synapseEnvironment;
+        this.vfsProperties = vfsProperties;
+    }
+
+    /**
+     * Inject the message to the sequence
+     */
+    public boolean invoke(Object object, String name) throws SynapseException {
+
+        ManagedDataSource dataSource = null;
+
+        FileObject file = (FileObject) object;
+        InputStream in = null;
+        try {
+            org.apache.synapse.MessageContext msgCtx = createMessageContext();
+            msgCtx.setProperty(SynapseConstants.INBOUND_ENDPOINT_NAME, name);
+            msgCtx.setProperty(SynapseConstants.ARTIFACT_NAME, SynapseConstants.FAIL_SAFE_MODE_INBOUND_ENDPOINT + name);
+            msgCtx.setProperty(SynapseConstants.IS_INBOUND, true);
+            if (vfsProperties.getReplyFileURI() != null && !vfsProperties.getReplyFileURI().isEmpty()) {
+                msgCtx.setProperty(Constants.OUT_TRANSPORT_INFO,
+                        new VFSOutTransportInfo(vfsProperties.getReplyFileURI(), vfsProperties.isFileLocking()));
+            }
```

Comment:
Is it working? If not, let's remove this.

Reply:
No, we need to remove it. This logic was added before we decided to handle the reply file.